### ISSUE TITLE
Slow performance after closing a ticket

### ISSUE THREAD

**Issue by sayajin101** [osTicket/osTicket#5365]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ ] Do you have any errors in the PHP error log, or javascript console?
* [x] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

[Description of the bug or feature]
After closing a ticket, this query is run for whatever reason.
The ost_ticket table contains almost 800,000 tickets.
`SELECT COUNT(DISTINCT CASE WHEN A1.`staff_id` IN (63) AND NOT A2.`id` IN (3) AND A1.`staff_id` IN (63) AND NOT A2.`id` IN (3) AND A1.`staff_id` IN (63) AND A2.`id` IN (1) AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q18`, COUNT(DISTINCT CASE WHEN A1.`staff_id` IN (63) AND NOT A2.`id` IN (3) AND A1.`staff_id` IN (63) AND NOT A2.`id` IN (3) AND NOT A1.`staff_id` IN (63) AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q19`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q1`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A2.`state` = 'open' AND A1.`isanswered` = '0' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q9`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A2.`state` = 'open' AND A1.`isanswered` = '1' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q5`, COUNT(DISTINCT CASE WHEN A2.`state` = 'open' AND A2.`state` = 'open' AND A1.`isoverdue` = '1' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q7`, COUNT(DISTINCT CASE WHEN A1.`staff_id` IN (63) AND NOT A2.`id` IN (3) AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q17`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q2`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2020-02-01 00:00:00' AND '2020-02-29 23:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q14`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2020-01-01 00:00:00' AND '2020-03-31 23:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q15`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2020-01-01 00:00:00' AND '2020-12-31 23:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q16`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2020-02-05 00:00:00' AND '2020-02-05 23:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q10`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2020-02-04 00:00:00' AND '2020-02-04 23:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q11`, COUNT(DISTINCT CASE WHEN A2.`state` = 'closed' AND A2.`state` = 'closed' AND A1.`closed` BETWEEN '2020-02-03 00:00:00' AND '2020-02-09 23:59:59' AND A3.`object_type` = 'T' THEN A1.`ticket_id` END) AS `q12` FROM `ost_ticket` A1 JOIN `ost_ticket_status` A2 ON (A1.`status_id` = A2.`id`) LEFT JOIN `ost_thread` A3 ON (A3.`object_type` = 'T' AND A1.`ticket_id` = A3.`object_id`) LEFT JOIN `ost_thread_referral` A4 ON (A3.`id` = A4.`thread_id`) LEFT JOIN `ost_staff` A5 ON (A4.`object_type` = 'S' AND A4.`object_id` = A5.`staff_id`) LEFT JOIN `ost_thread` A6 ON (A1.`ticket_id` = A6.`object_id` AND A6.`object_type` = 'C') LEFT JOIN `ost_thread_referral` A7 ON (A6.`id` = A7.`thread_id`) LEFT JOIN `ost_staff` A8 ON (A7.`object_type` = 'S' AND A7.`object_id` = A8.`staff_id`) LEFT JOIN `ost_department` B0 ON (A4.`object_type` = 'D' AND A4.`object_id` = B0.`id`) LEFT JOIN `ost_department` B1 ON (A7.`object_type` = 'D' AND A7.`object_id` = B1.`id`) WHERE ((A2.`state` = 'open' AND (A1.`staff_id` = 63 OR A5.`staff_id` = 63 OR A6.`object_type` = 'C' AND A8.`staff_id` = 63)) OR (A2.`state` IN ('open', 'closed') AND (A1.`dept_id` IN (1, 2, 3) OR B0.`id` IN (1, 2, 3))) OR A6.`object_type` = 'C' AND B1.`id` IN (1, 2, 3))`

### Steps to Reproduce

1. Close any ticket on the system

**Expected behavior:** Redirect to next page in a timely manner

**Actual behavior:** Takes about 13 seconds to load the next page after closing

### Versions

Admin panel -> Dashboard -> Information which also additionally gives you information about your server.
   12 CPU Cores
   16 Gigs of RAM
   CentOS Linux release 7.7.1908 (Core)
   mysql  Ver 15.1 Distrib 10.3.22-MariaDB, for Linux (x86_64) using readline 5.1
   PHP 7.3.14 (cli) (built: Jan 21 2020 09:59:50) ( NTS )
   OS Ticket v1.14.1

Also, please include the OS and what version of the OS you're running. As well as your browser and browser version.
   Windows 10
   Chrome Version 79.0.3945.130 (Official Build) (64-bit)

**Comment by JediKev** [comment]
@sayajin101 

What is your resource usage like when that query runs? Does your CPU usage or RAM usage spike really high?

Cheers.

**Comment by sayajin101** [comment]
The 1 CPU core maxes out at 100%, the reason this is causing such an issue is because it has to complete the query before it loads the tickets page after closing a ticket.

So the client waits for about 13+ seconds to a page to load

**Comment by JediKev** [comment]
@sayajin101 

Right, have you tried adding another CPU or upgrading the one you currently have?

Cheers.
